aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2543,59 +2577,17 @@ static Tensor& linalg_norm_out_impl(Tensor& result, const Tensor& self, const op
   return result;
 }
 
-static Tensor& linalg_vector_norm_impl(const Tensor& self, const Scalar& scalar_ord, OptionalIntArrayRef opt_dim, bool keepdim, optional<ScalarType> opt_dtype, Tensor& result) {
-  // Casting a large integer to a double will introduce some error, but for
-  // practical purposes, it won't matter since a large order will usually
-  // give an infinite result
+TORCH_IMPL_FUNC(linalg_vector_norm_out)(const Tensor& self, const Scalar& scalar_ord, OptionalIntArrayRef opt_dim, bool keepdim, optional<ScalarType> opt_dtype, const Tensor& result) {
+  // Casting a large integer to a double will just introduce an error for
+  // values larger than 10^53 (same for negative numbers), so that's fine.
   auto ord = scalar_ord.toDouble();
+  auto dim = opt_dim.value_or(IntArrayRef{});
+  auto in_dtype = opt_dtype.value_or(self.scalar_type());
 
-  TORCH_CHECK(self.device().type() == DeviceType::CPU || self.device().type() == DeviceType::CUDA,
-              "linalg.vector_norm only supports CPU and CUDA device types, but got: ",
-              self.device().type());
-  TORCH_CHECK(self.layout() == Layout::Strided,
-              "linalg.vector_norm only supports strided layout, but got: ", self.layout());
-
-  if (opt_dtype.has_value() && isComplexType(self.scalar_type())) {
-    TORCH_CHECK(isComplexType(opt_dtype.value()),
-      "linalg.vector_norm expected complex 'dtype', since input is complex, ",
-      "but got ", opt_dtype.value());
-  }
-
-  checkFloatingOrComplex(self, "linalg.vector_norm");
-  ScalarType in_dtype = opt_dtype.value_or(self.scalar_type());
-
-  IntArrayRef dim = opt_dim.value_or(IntArrayRef{});
-
-  if (self.numel() == 0) {
-    // TODO: The question about how to handle negative orders when the input
-    // is empty has not been settled yet. For now, we raise an error. Issue:
-    // https://github.com/pytorch/pytorch/issues/52783
-    TORCH_CHECK(ord >= 0,
-      "linalg.vector_norm of negative order cannot be performed on an empty tensor");
-
-    // For NumPy compatibility, we can only perform order infinity reduction
-    // (max/min) on a tensor with zero elements if the dimensions to reduce are
-    // nonzero. Otherwise, throw an error.
-    if (ord == INFINITY) {
-      bool has_identity = true;
-
-      if (dim.size() == 0) {
-        has_identity = false;
-      } else {
-        for (int64_t dim_num : dim) {
-          if (self.size(dim_num) == 0) {
-            has_identity = false;
-            break;
-          }
-        }
-      }
-      TORCH_CHECK(has_identity,
-        "linalg.vector_norm cannot compute the infinity norm on an empty ",
-        "dimension because the operation does not have an identity");
-    }
-  }
   Tensor self_;
-  if (self.device().type() == c10::kCPU && isComplexType(self.scalar_type()) && std::abs(ord) == INFINITY) {
+  if (self.device().type() == c10::kCPU &&
+      isComplexType(self.scalar_type()) &&
+      std::abs(ord) == INFINITY) {
```

Comment:
I tried to get rid of this logic by using `norm_stub`, but no dice. The gradchecks fail if I remove it. This is a pre-existing condition that affects also `torch.norm`, but interestingly enough, `torch.norm` was not tested on inputs with `ord=inf`  and `-inf`. gradgrad fails (and the noncontiguous test as well) fails on CPU if I add those samples to the OpInfo. Should I investigate further all these issues?

Reply:
Yeah that's the difference between vectorized and non-vectorized implementations, we had discussions about it. We can leave it.

Reply:
I left a comment for future generations.